ACM-18617: Add Image-based Break/Fix to Dev-Preview Documentation (#26)
stolostron/stolostron: dev-preview/README.md

- **Virtualization/VM Level**: Focuses on resource insights at the VM level.

To understand how this can help you optimize resources across multiple clusters, as well as learn about installation steps, detailed usage, and how to start using the feature, you can refer to the full documentation [here](https://github.com/stolostron/right-sizing/blob/main/README.md).

## Image-Based Break/Fix (IBBF) for SNO Hardware Replacement

> The Image-Based Break/Fix (IBBF) feature streamlines Single Node OpenShift (SNO) hardware replacement, minimizing downtime while preserving the cluster’s original identity. It ensures that critical details, including cluster identifiers, cryptographic keys (such as kubeconfig), and authentication credentials, are retained. This allows the replacement node to seamlessly assume the identity of the failed hardware.

Reviewer: These are just suggestions to make it a bit easier to understand and accessible.
```suggestion
The Image-Based Break/Fix (IBBF) feature streamlines singe-node OpenShift (SNO) hardware replacement by minimizing downtime while preserving the cluster’s original identity. The IBBF feature ensures that critical details are retained, including cluster identifiers, cryptographic keys, such as kubeconfig, and authentication credentials. The preservation of critical details allows the replacement node to seamlessly assume the identity of the failed hardware.
```

> Designed for like-for-like hardware replacements and IBI-installed SNOs, IBBF introduces a GitOps-compatible, declarative API, enabling users to initiate hardware replacement with a single Git commit. Powered by the SiteConfig Operator and IBI Operator, this functionality allows clusters to be redeployed using the existing ClusterInstance CR. The SiteConfig Operator provides a flexible reinstallation service that integrates seamlessly with template-driven provisioning. Additionally, it includes a backup and restore mechanism for Secrets and ConfigMaps, ensuring the cluster retains its identity.

Reviewer: I don't remember if it's the SiteConfig Operator that has the backup and restore mechanism but I'd clarify what is "it".
```suggestion
Designed for like-for-like hardware replacements and IBI-installed SNOs, IBBF introduces a GitOps-compatible, declarative API that enables users to initiate hardware replacement with a single Git commit. Powered by the SiteConfig Operator and IBI Operator, this functionality allows clusters to be redeployed by using the existing ClusterInstance CR. The SiteConfig Operator provides a flexible reinstallation service that integrates seamlessly with template-driven provisioning. Additionally, the SiteConfig Operator includes a backup and restore mechanism for Secrets and ConfigMaps, ensuring the cluster retains its identity.
```

With IBBF, OpenShift users gain a resilient, automated, and GitOps-native solution for rapidly restoring SNO clusters after hardware failures.

> Additionally, the SiteConfig Operator extends its reinstallation service to clusters provisioned via the Assisted Installer. However, unlike IBBF, where the cluster identity is preserved, Assisted Installer-based reinstallations generate a new cluster identity during the process.

Reviewer: I don't think we need that "Additionally" here but I'll leave it up to you!
```suggestion
The SiteConfig Operator extends its reinstallation service to clusters provisioned with the Assisted Installer. However, unlike IBBF, where the cluster identity is preserved, Assisted Installer-based reinstallations generate a new cluster identity during the process.
```
